proxy-node-translator/Dockerfile

```diff
@@ -12,41 +9,20 @@ COPY build.gradle build.gradle
 COPY settings.gradle settings.gradle
 COPY proxy-node-shared/ proxy-node-shared/
 
-ARG component=proxy-node-translator
+ARG component=proxy-node-gateway
 COPY ${component}/src ${component}/src
 COPY ${component}/build.gradle ${component}/build.gradle
 
 RUN gradle --no-daemon --quiet -p ${component} --parallel install -x test
 ENTRYPOINT ["gradle", "--no-daemon"]
 
-# -------------------- Runtime Image --------------------
 FROM openjdk:11-jre-slim
-
-# ----------------------------------------------------------
-# Install softHSM and dependencies for OpenSC
-RUN apt-get update && apt-get install -y softhsm pcscd libccid libpcsclite-dev libssl-dev libreadline-dev autoconf automake build-essential docbook-xsl xsltproc libtool pkg-config wget
-
-# Download and build OpenSC
-ENV OPEN_SC_VERSION=0.19.0
-RUN wget https://github.com/OpenSC/OpenSC/releases/download/${OPEN_SC_VERSION}/opensc-${OPEN_SC_VERSION}.tar.gz && \
-    tar xfvz opensc-${OPEN_SC_VERSION}.tar.gz && \
-    cd opensc-${OPEN_SC_VERSION} && \
-    ./bootstrap && ./configure --prefix=/usr --sysconfdir=/etc/opensc && \
-    make && make install && \
-    cd .. && rm -rf opensc* && \
-    rm -rf /var/lib/apt/lists/*
-
-# ----------------------------------------------------------
-
-# Copy java app from build to runtime image
 WORKDIR /app
 USER root
 
-ARG component=proxy-node-translator
+ARG component
 COPY --from=build /app/${component}/build/install/${component} /app
-COPY --from=build /app/${component}/src/dist/init.sh /bin/init.sh
 
 ENV CONFIG_FILE config.yml
 ENV COMPONENT $component
-ENTRYPOINT ["/bin/init.sh"]
-CMD []
+CMD "bin/$COMPONENT"
```

Comment:
Is there a reason why you want to pass `bin/$COMPONENT` as a command to whatever the base image's `ENTRYPOINT` is rather than running the process directly?

This can have impacts on how signals are propagated to the process and can make it harder to reason about the process tree within the container. I found it's good practice to set the desired `ENTRYPOINT` to and clear out any `CMD`from the parent image to make it very explict what you are running.

Reply:
Yeah, I realised signals weren't being handled properly so I'm trying `dumb-init` for the `ENTRYPOINT` and leaving the `CMD` as-is. The main reason was so one could debug the container using `docker run` without having to override the entrypoint.